src/app.ts

```diff
@@ -13,14 +13,45 @@ interface TemperatureSummary {
   average: number
 }
 
+const processedReadings: TemperatureReading[] = [];
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  processedReadings.push(...readings);
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  let filteredReadings: TemperatureReading[] = [];
+
+  filteredReadings = processedReadings.filter(
+    (read) => (read.time.getTime() === date.getTime()) && (read.city === city),
+  )
+
+  if (filteredReadings.length !== 0) {
```

Comment:
Since else is only returning null, we can do fail fast here, also you can use (!filteredReadings.length)